cylc/flow/config.py

```diff
@@ -925,7 +925,7 @@ def _check_circular_helper(x2ys, y2xs):
         """
         # Starter x nodes are those with no incoming, i.e.
         # x nodes that never appear as a y.
-        sxs = set(x01 for x01 in x2ys if x01 not in y2xs)
+        sxs = {x01 for x01 in x2ys if x01 not in y2xs}
```

Comment:
While we're at it, this feels more elegant., and performance is slightly better from a couple of tests I've tried
```suggestion
        sxs = set(x2ys).difference(y2xs)
```